agnpy/spectra/spectra.py

```diff
@@ -473,3 +477,37 @@ def __str__(self):
             + f" - gamma_min: {self.gamma_min:.2e}\n"
             + f" - gamma_max: {self.gamma_max:.2e}"
         )
+
+
+class Interpolation(ElectronDistribution):
+
+    def __init__(
+        self,
+        gamma_data,
+        n_data
+    ):
+        self.gamma_data = gamma_data
+        self.n_data = n_data.value
+        self.gamma_max = max(self.gamma_data)
+        self.gamma_min = min(self.gamma_data)
+        self.f_log = CubicSpline(np.log10(self.gamma_data), np.log10(self.n_data))
+
+    #not a static method, since there's the self.f_log inside
+    #Had to be included, since CubicSpline extrapolates beyong gamma min and gamma max
+    def evaluate(self, gamma, gamma_min, gamma_max):
+        return np.where(
+            (gamma_min <= gamma) * (gamma <= gamma_max),
+            np.power(10,self.f_log(np.log10(gamma)))* u.Unit('cm-3'),
+            0
+        )
+
+    def __call__(self, gamma):
+        return self.evaluate(
+            gamma, self.gamma_min, self.gamma_max
+        )
+
+    #deriv is the derivative of 10**(f_log(np.log10(gamma))). Chain rule is used
+    def SSA_integrand(self, gamma):
```

Comment:
can you double-check the formula? This is what I obtain and I see that at least a ln(10) is missing

![CodeCogsEqn](https://user-images.githubusercontent.com/26409711/195786068-35867986-8a3c-4c20-86b0-13576e1c0a67.png)

Also why are you using each time `np.power(10,self.f_log(np.log10(gamma)))` to evaluate the actual function, when you have that defined in `evaluate` already?